cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -101,4 +101,16 @@ type NodeConfiguration struct {
 	NodeName                 string
 	TLSBootstrapToken        string
 	Token                    string
+
+	// TLSDiscoveryRootCAPubKeys specifies a set of public key pins. The root CA
+	// found during TLS discovery must match one of these values. Specifying an
+	// empty set disables root CA pinning. Each hash is a hex encoded SHA-256
+	// hash of the Subject Public Key Info (SPKI) object in DER-encoded ASN.1.
+	// This is similar to the format defined in https://tools.ietf.org/html/rfc7469#section-2.4.
+	// These hashes can be calculated using, e.g.,
+	// `openssl x509 -pubkey -in ca.crt openssl rsa -pubin -outform der 2>&/dev/null | openssl dgst -sha256 -hex`
+	TLSDiscoveryRootCAPubKeys []string
```

Comment:
Some problems with this:
* There is no indication that this is supposed to be a hash. Preferably the name should carry this indication.
* There is no indication that this is only relevant for token discovery. Preferably the name should carry this indication.
* We should namespace hashes so we can support multiple hashes incase we need to move away from sha256 for whatever reason.
* There is no indication of the format of the hash serialization. Does hex work? Does base64 work? It's ok to note this in the field documentation.

Strawman:
* Change name to `DiscoveryTokenCACertHashes`
* Expect and document a serialization format of `<hash-type>:<base16-encoded-sum>` and a list of supported hashes starting with `sha256` only